WordPress/gutenberg
# Deprecate 'Post author' block

## What?
Closes #53427

This PR deprecates the Post Author block (`core/post-author`) and provides a means to convert it to a recommended related block.

From my understanding, deprecating a block means that **we will no longer actively fix bugs or add features**. In that case, we should be able to close the following issues/PRs as well.

- #24952
- #29450
- #35692
- #35693
- #35694
- #35964
- #37748
- #40380
- #47169
- #47211
- #53561
- #53687

## Why?

All elements of the Post Author block are fixed. Furthermore, although it has block support, it is not possible to apply styles to those elements individually. This block should be able to be completely replaced by the Avatar block (`core/avatar`), Author Name block (`core/post-author-name`), Author Biography block (`core/post-author-biography`) and Paragraph block (`core/paragraph`).

## How?

First, please see the following diagram to see how and to which block the attributes of this block are transformed. While maintaining all block support, the use of Row/Stack provides a transformation that preserves the current layout.

![Deprecate Post Author Block](https://github.com/WordPress/gutenberg/assets/54422211/0d2bbcad-159a-44ea-9f85-4e57c9665789)

The user can migrate to the recommended block via the group block that appears in the conversion menu on the block toolbar.

However, since this action essentially means "wrapping" the current block with a group block, the user may not realize that this is meant to be a migration to a recommended block.

Therefore, I have further added a button to the block sidebar for migration.

![352894042-6857ec94-9808-4a8c-a2f7-6053fae3746e](https://github.com/user-attachments/assets/c8b19be4-e79c-4bfb-8b3b-fbc382e6b5bc)

## Testing Instructions

- Paste the HTML below into the code editor to insert the block.
  ```html
  
  ```
- Change styles and settings from the block toolbar and block sidebar.
- Try transforming from the "Migrate block" button on the block toolbar or block sidebar.
- It should convert to the recommended block without much change in style, settings, or layout.

## Screenshots or screencast 

https://github.com/user-attachments/assets/990c4fed-b9e9-4674-8e84-cbf769ca0152
